src/transactions/ParallelApplyUtils.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright 2025 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "transactions/ParallelApplyUtils.h"
+#include "transactions/MutableTransactionResult.h"
+
+namespace stellar
+{
+
+std::unordered_set<LedgerKey>
+getReadWriteKeysForStage(ApplyStage const& stage)
+{
+    std::unordered_set<LedgerKey> res;
+
+    for (auto const& txBundle : stage)
+    {
+        for (auto const& lk :
+             txBundle.getTx()->sorobanResources().footprint.readWrite)
+        {
+            res.emplace(lk);
+            if (isSorobanEntry(lk))
+            {
+                res.emplace(getTTLKey(lk));
```

Comment:
Do we actually care about TTL entries for the purpose of this set? It doesn't seem like we ever actually test TTLs for  being present in the set.

Reply:
We use the result of this function in `writeDirtyMapEntriesToLedgerTxn` so we can do a direct lookup of the TTL.